js/app.js

```diff
@@ -0,0 +1,252 @@
+    return `${city.name} is in ${city.country} and has an area of ${city.areaInKilometers} square kilometers`;
+} // so I understand that $ before {} gives the possibility to target a property form an object
+
+console.log(getCityInformation({
+    name: 'Warsaw',
+    country: 'Poland',
+    areaInKilometers: 517
+})); // Output: Warsaw is in Poland and has an area of 517 square kilometers
+
+// 9
+function getUserCopy(user9) {
+    return {
+        firstName: user9.firstName,
+        lastName: user9.lastName
+    } ;
+}
+
+const john = {
+    firstName: 'John',
+    lastName: 'Smith'
+}
+
+const newUser = getUserCopy(john);
+console.log(newUser.firstName); // John
+console.log(newUser.lastName); // Smith
+console.log(newUser === john); // false <-- this is crucial
+
+// 10 identical as the warsaw assignment
+function animal(obj) {
+    return `This ${obj.color} ${obj.name} has ${obj.legs} legs.`;
+}
+
+console.log(animal({ name: "dog", legs: 4, color: "white" })); // Output: "This white dog has 4 legs."
+
+// 11 only added comma's and closed some }
+var rooms = {
+    first: {
+        description: 'This is the first room',
+        items: {
+            chair: 'The old chair looks comfortable',
+            lamp: 'This lamp looks ancient',
+        }
+    },
+    second: {
+        description: 'This is the second room',
+        items: {
+            couch: 'This couch looks like it would hurt your back',
+            table: 'On the table there is an unopened bottle of water',
+        }
+    }
+}
+
+// 12
+/*
+function addContact(contacts, name, phoneNumber) {
+    const newContacts = {};
+
+    newContacts[name] = {
+        name: name,
+        phoneNumber: phoneNumber,
+    };
+    return newContacts;
+}
+
+const contacts = {};
+const contactsWithJohn = addContact(contacts, 'John', '123 456 789');
+
+console.log(contactsWithJohn.John.name); // "John"
+console.log(contactsWithJohn.John.phoneNumber); // '123 456 789'
+
+console.log(contacts === contactsWithJohn); // false
+console.log(contacts.John); // undefined
+*/
+
+// 13
+function addContact(contacts, name, phoneNumber) {
+    const newContacts = {};
+
+    newContacts[name] = {
+        name: name,
+        phoneNumber: phoneNumber,
+    };
+    return newContacts;
+}
+
+function findContact(contacts, name) {
+    if (contacts[name]) {
+        return `Name: ${contacts[name].name}, Phone: ${contacts[name].phoneNumber}`;
+    }
+    return undefined;
+}
+
+const contacts = {};
+const contactsWithJohn = addContact(contacts, 'John', '123 456 789');
+
+const johnContactInfo = findContact(contactsWithJohn, 'John');
+console.log(johnContactInfo); // "Name: John, Phone: 123 456 789"
+
+// 14
+function validateIfObjectIsAUser(object) {
+    if (
+        typeof object.firstName === 'string' &&
+        typeof object.lastName === 'string' &&
+        typeof object.age === 'number' &&
+        object.greet() === `Hi, I'm ${object.firstName} ${object.lastName}!`
```

Comment:
> returns a string containing the user firstName and the user lastName

Please make it more generic. Just check if the returned string contains both first and last name